app/furniture/marketplace/cart_products_controller.rb

```diff
@@ -55,6 +57,7 @@ def update
     end
 
     def destroy
+      authorize(cart_product)
```

Comment:
Could we do this in the `cart_product` method body itself, so that we authorize each time?